module/Frontpage/config/module.config.php

```diff
@@ -0,0 +1,43 @@
+<?php
+return array(
+    'controllers' => array(
+        'invokables' => array(
+            'Frontpage\Controller\Frontpage' => 'Frontpage\Controller\FrontpageController',
+        ),
+    ),
+    'router' => array(
+        'routes' => array(
+            'home' => array(
+                'type'    => 'Literal',
+                'options' => array(
+                    'route'    => '/',
+                    'defaults' => array(
+                        '__NAMESPACE__' => 'Frontpage\Controller',
+                        'controller'    => 'Frontpage',
+                        'action'        => 'home',
+                    ),
+                ),
+                'may_terminate' => true,
+                'child_routes' => array(
+                    'page' => array(
+                        'type'    => 'Segment',
+                        'options' => array(
+                            'route'    => '/[:category[/:name]]',
```

Comment:
Since this is a child route of `/`, it shouldn't start with a `/`. Also, the category wouldn't be optional for this route. And defaults should be created for the optional part. And a controller and action still have to be determined.

Also, since it kinda is a catch-all route, you should set a priority.